src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,100 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final int INCREASE_CAPACITY = 2;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private int size;
+    private int capacity;
+    private int threshold;
+    private Node<K, V>[] nodes;
+
+    public MyHashMap() {
+        capacity = INITIAL_CAPACITY;
+        threshold = (int) (INCREASE_CAPACITY * DEFAULT_LOAD_FACTOR);
+        nodes = new Node[capacity];
+    }
+
+    private static class Node<K, V> {
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkSize();
+        Node<K, V> newNode = new Node<>(key, value, null);
+        int hashPosition = getNodeIndex(key, capacity);
+        Node<K, V> positionNode = nodes[hashPosition];
+        if (positionNode == null) {
+            nodes[hashPosition] = newNode;
+            size++;
+            return;
+        }
+        while (positionNode != null) {
+            if ((key == positionNode.key) || (key != null && key.equals(positionNode.key))) {
+                positionNode.value = value;
+                return;
+            }
+            if (positionNode.next == null) {
+                positionNode.next = newNode;
+                size++;
+                return;
+            }
+            positionNode = positionNode.next;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> getNode = nodes[getNodeIndex(key, capacity)];
```

Comment:
```suggestion
        Node<K, V> node = nodes[getNodeIndex(key)];
```
rename this variable to **node** in other rows